lib/rucio/richclient.py

```diff
@@ -0,0 +1,302 @@
+        'GROUP': 'medium_purple3',
+        'SERVICE': 'yellow'
+    }
+
+
+def setup_rich_logger(
+    module_name: Optional[str] = None,
+    logger_name: Optional[str] = None,
+    logger_level: Optional[int] = None,
+    verbose: bool = False,
+    console: Optional[Console] = None
+) -> logging.Logger:
+    """
+    Factory method to set logger with RichHandler.
+
+    The function is a copy of the method in rucio.common.utils setup_logger() with minor changes.
+
+    :param module_name: __name__ of the module that is calling this method
+    :param logger_name: name of the logger, typically name of the module.
+    :param logger_level: if not given, fetched from config.
+    :param verbose: verbose option set in bin/rucio
+    :param console: Rich console object
+    :returns: logger with RichHandler
+    """
+    # Helper method for cfg check.
+    def _force_cfg_log_level(cfg_option: str) -> bool:
+        cfg_forced_modules = config_get('logging', cfg_option, raise_exception=False, default=None, clean_cached=True, check_config_table=False)
+        if cfg_forced_modules and module_name is not None:
+            if re.match(str(cfg_forced_modules), module_name):
+                return True
+        return False
+
+    if not logger_name:
+        if not module_name:
+            logger_name = 'user'
+        else:
+            logger_name = module_name.split('.')[-1]
+    logger = logging.getLogger(logger_name)
+
+    # Extracting the log level.
+    if not logger_level:
+        logger_level = logging.INFO
+        if verbose:
+            logger_level = logging.DEBUG
+
+        # Overriding by the config.
+        cfg_levels = (logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR)
+        for level in cfg_levels:
+            cfg_opt = 'forceloglevel' + logging.getLevelName(level)
+            if _force_cfg_log_level(cfg_opt):
+                logger_level = level
+
+    logger.setLevel(logger_level)
+
+    def add_handler(logger: logging.Logger,
+                    console: Optional[Console] = None,
+                    verbose: bool = False) -> None:
+
+        def time_formatter(timestamp: datetime) -> Text:
+            return Text(f"[{timestamp.isoformat(sep=' ', timespec='milliseconds')}]")
+
+        console = console or Console()
+        handler = RichHandler(rich_tracebacks=True, markup=True, show_path=verbose, show_time=verbose, console=console, tracebacks_width=min(console.width, MAX_TRACEBACK_WIDTH),
+                              tracebacks_word_wrap=True, log_time_format=time_formatter)
+        logger.addHandler(handler)
+
+    # Setting handler and formatter.
+    if not logger.handlers:
+        add_handler(logger, console, verbose)
+
+    return logger
+
+
+def generate_table(
+    rows: Sequence[Sequence[RenderableType]],
+    headers: Optional[Sequence[RenderableType]] = None,
+    row_styles: Optional[Sequence[StyleType]] = None,
+    col_alignments: Optional[list[JustifyMethod]] = None,
```

Comment:
Is there a rhyme or reason to the left/right alignments? It may be cleaner to automatically assign that based on the first row's numeric/not numeric status for each column

Reply:
I included a line, **231**,  that does automatic assignment for the simple cases 
```python
col_alignments = col_alignments or ['right' if str(col).isnumeric() else 'left' for col in rows[0]]
```

However, this approach does not currently work properly in all cases. For example:
- File sizes in human-readable formats (e.g., `100 MB`) are not handled correctly.
- If the first row in a column contains `None`, even though other rows are numerical, it will default to left alignment.
- For tables where columns have mixed value types (e.g., the two column key-value pair tables with different types), it's better to manually assign alignments.

Hence, the use of manual left/right alignments for now.